helix-view/src/document.rs

```diff
@@ -265,6 +276,174 @@ impl Document {
         }
     }
 
+    fn detect_indent_style(&mut self) {
+        // Determine whether a character is a line break.
+        //
+        // TODO: this is probably a generally useful utility function.  Where
+        // should we put it?
+        fn char_is_linebreak(c: char) -> bool {
+            [
+                '\u{000A}', // LineFeed
+                '\u{000B}', // VerticalTab
+                '\u{000C}', // FormFeed
+                '\u{000D}', // CarriageReturn
+                '\u{0085}', // NextLine
+                '\u{2028}', // Line Separator
+                '\u{2029}', // ParagraphSeparator
+            ]
+            .contains(&c)
```

Comment:
I think a `matches!(c, a | b | c | d | e)` is more efficient (it uses a `match` expression)

Reply:
Yeah, originally it was a standard `match` statement, and then clippy warned to use `matches!`.  I tried to do that, and if I get rid of the comments that works.  But the macro chokes and fails to parse on the per-line items with comments.  So I ended up with this just to shut clippy up, ha ha.

I guess I can just remove the comments, but it seems a little obtuse that way.

Reply:
Whoops, never mind.  Tried it again and it's fine.  I must have done something stupid last time.